Mathlib/LinearAlgebra/Projection.lean

```diff
@@ -424,6 +425,41 @@ theorem eq_conj_prod_map' {f : E →ₗ[R] E} (h : IsProj p f) :
   · simp only [coe_prodEquivOfIsCompl, comp_apply, coe_inr, coprod_apply, map_zero,
       coe_subtype, zero_add, map_coe_ker, prodMap_apply, zero_apply, add_zero]
 
+theorem isIdempotentElem {R V : Type*} [Semiring R] [AddCommMonoid V] [Module R V]
+    {T : V →ₗ[R] V} {U : Submodule R V} (h : IsProj U T) : IsIdempotentElem T := by
+  ext; exact h.2 _ (h.1 _)
+
+protected theorem range
+    {R V : Type*} [Semiring R] [AddCommMonoid V] [Module R V] {T : V →ₗ[R] V} {U : Submodule R V}
+    (h : IsProj U T) : range T = U := by
+  ext x
+  refine ⟨fun ⟨y, hy⟩ => ?_, fun hx => ⟨x, h.map_id x hx⟩⟩
+  rw [← hy]
+  exact h.map_mem y
```

Comment:
This is more a matter of personal style, but if you want, there is also the option of a term-mode rewrite:
```suggestion
  exact ⟨fun ⟨y, hy⟩ => hy ▸ h.map_mem y, fun hx => ⟨x, h.map_id x hx⟩⟩
```